lib/parsers/iso_base_media_file_format/utils.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'matrix'
+
+module FormatParser::ISOBaseMediaFileFormat::Utils
+  def codecs(box_tree)
+    video_trak_boxes(box_tree).map do |trak_box|
+      stsd_box = trak_box['mdia']['minf']['stbl']['stsd']
```

Comment:
Are we sure about the existence of all these keys, or shall we rather `dig` and handle `nil`s?

Reply:
All of these boxes are specified as mandatory in the ISOBMFF spec, so if they don't exist then that indicates that the file is invalid or that we have a bug elsewhere in the code - both of which warrant an error being thrown imo.